utils/forest_utils/src/io/progress_bar.rs

```diff
@@ -1,11 +1,22 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
+// JANK(aatifsyed): I don't really understand why this module exists, a lot of
```

Comment:
If I recall correctly, it exists to not use `pbr` or other progress bar crates directly but to have it abstracted. This way, if/when our progress bar backend becomes abandonware or there's a new cool kid on crates.io, we don't have to change it everywhere. Plus, we ensure it's uniform - we don't want one part of the code to use one type of progress bar and another to use something else.

Reply:
That's an odd decision to me - we don't do that kind of wrapping for our other dependencies like dialoguer, and if we wanted to, we should use `impl Trait`. 

Agreed on uniformity though :P

Reply:
We should do this for dialoguer as well. The way we do it is a detail, I'm fine with either approach. What we want to achieve is a uniform UI.

Reply:
#2928